core/services/prom_reporter.go

```diff
@@ -97,6 +125,36 @@ func (pr *promReporter) Disconnect() {
 }
 
 func (pr *promReporter) OnNewLongestChain(ctx context.Context, head models.Head) {
+	pr.newHeads.Deliver(head)
+}
+
+func (pr *promReporter) eventLoop() {
+	logger.Debug("PromReporter: starting event loop")
+	defer pr.wgDone.Done()
+
+	for {
+		select {
+		case <-pr.newHeads.Notify():
+			item, exists := pr.newHeads.Retrieve()
+			if !exists {
+				continue
+			}
+			head, ok := item.(models.Head)
+			if !ok {
+				logger.Errorf("expected `models.Head`, got %T", item)
+				continue
+			}
+			pr.reportMetrics(head)
+
+		case <-pr.chStop:
+			return
+		}
+	}
+}
+
+func (pr *promReporter) reportMetrics(head models.Head) {
+
+	ctx := context.Background()
```

Comment:
Perhaps we could use the passed in context?

Reply:
yup, passed a context from chStop now